tests/pre-exit-report-link.bats

```diff
@@ -0,0 +1,150 @@
+  stub curl "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success'"
+  stub buildkite-agent "annotate --style info --context \"test-collector\"  \* : echo 'annotation success'"
+  stub which "jq : exit 1"
+  
+  run "$PWD/hooks/pre-exit"
+
+  unstub which
+  unstub buildkite-agent
+  unstub curl
+
+  assert_success
+  assert_output --partial "Uploading './tests/fixtures/junit-1.xml'..."
+  assert_output --partial "curl success"
+  assert_output --partial "Not using jq"
+  assert_output --partial "annotation success"
+}
+
+@test "Annotates report link from multiple file and same report URLs" {
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_FILES='**/*/junit-*.xml'
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_ANNOTATE_LINK="true"
+  export CURL_RESP_FILE="./tests/fixtures/response.json"
+
+  stub curl \
+    "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success 1'" \
+    "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success 2'" \
+    "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success 3'"
+  stub buildkite-agent "annotate --style info --context \"test-collector\"  \* : echo 'annotation success'"
+  
+
+  run "$PWD/hooks/pre-exit"
+
+  unstub buildkite-agent
+  unstub curl
+
+  assert_success
+  assert_output --partial "Uploading './tests/fixtures/junit-1.xml'..."
+  assert_output --partial "Uploading './tests/fixtures/junit-2.xml'..."
+  assert_output --partial "curl success 1"
+  assert_output --partial "curl success 3"
+  assert_output --partial "Got 1 report URLs."
+  assert_output --partial "annotation success"
+}
+
+@test "Annotates report link from multiple file and different report URLs" {
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_FILES='**/*/junit-*.xml'
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_ANNOTATE_LINK="true"
+  export CURL_RESP_FILE="./tests/fixtures/response.json"
+
+  stub curl \
+    "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success 1'" \
+    "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success 2'" \
+    "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success 3'"
+  stub jq \
+    "-r '.run_url' \* : echo https://buildkite.com/organizations/example/analytics/suites/collector-test/runs/1" \
+    "-r '.run_url' \* : echo https://buildkite.com/organizations/example/analytics/suites/collector-test/runs/2" \
+    "-r '.run_url' \* : echo https://buildkite.com/organizations/example/analytics/suites/collector-test/runs/1"
+  stub buildkite-agent "annotate --style info --context \"test-collector\"  \* : echo 'annotation success'"
```

Comment:
I would at least do an `echo 'annotation success' with params \${6}` so that you can later assert that the annotation contains both links

Reply:
I could not made this change work. Sorry. But the test is already validating the number of links contained:
```
assert_output --partial "Got 2 report URLs."
```

Reply:
If you leave the stub as I suggested:
```
  stub buildkite-agent "annotate --style info --context \"test-collector\"  \* : echo 'annotation success' with params \${6}"
```

and then add the following assertions:
```
  assert_output --partial "- [Report #1](https://buildkite.com/organizations/example/analytics/suites/collector-test/runs/1)"
  assert_output --partial "- [Report #2](https://buildkite.com/organizations/example/analytics/suites/collector-test/runs/2)"
  refute_output --partial "- [Report #3]"
```

The test should pass as expected :)

But as it is right now you are only cheking that there are 2 report URLs, not what they are... which actually unveiled a bug for me